src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -1042,7 +1041,7 @@ private void ReadCompressedTextChunk(ImageMetadata baseMetadata, PngMetadata met
             }
 
             int zeroIndex = data.IndexOf((byte)0);
-            if (zeroIndex < PngConstants.MinTextKeywordLength || zeroIndex > PngConstants.MaxTextKeywordLength)
+            if (zeroIndex is < PngConstants.MinTextKeywordLength or > PngConstants.MaxTextKeywordLength)
```

Comment:
Just FYI: https://github.com/dotnet/roslyn/issues/60534

I'd leave the code as is, as it's basically the same as now just nicer to read. When Roslyn got that addressed the perf-goodness comes "for free".